source/dub/generators/generator.d

```diff
@@ -383,36 +383,30 @@ class ProjectGenerator
 		}
 
 		// 5. override string import files in dependencies
-		static void overrideStringImports(ref TargetInfo ti, TargetInfo[string] targets, string[] overrides)
+		static void overrideStringImports(ref TargetInfo ti, TargetInfo* pti, TargetInfo[string] targets, string[] overrides)
 		{
 			// do not use visited here as string imports can be overridden by *any* parent
 			//
 			// special support for overriding string imports in parent packages
 			// this is a candidate for deprecation, once an alternative approach
 			// has been found
-			if (ti.buildSettings.stringImportPaths.length) {
+			if (pti && ti.buildSettings.stringImportPaths.length) {
 				// override string import files (used for up to date checking)
-				foreach (ref f; ti.buildSettings.stringImportFiles)
+				if (ti.buildSettings.stringImportFiles
+					.map!(f => NativePath(f).head)
+					.any!(f => overrides.any!(of => NativePath(of).head == f)))
```

Comment:
I would split this change, as it looks like pure refactoring, in a separate commit.